File: bob.py
import subprocess
import json

def renderizar_sudoku(grid):
    print("\n--- Sudoku 25x25 Resuelto ---")
    for i in range(25):
        if i % 5 == 0 and i != 0:
            print("-" * 75)
        fila = []
        for j in range(25):
            if j % 5 == 0 and j != 0:
                fila.append("|")
            val = grid[i][j]
            fila.append(str(val).rjust(2) if val != 0 else ".")
        print(" ".join(fila))

# 1. Crear tablero inicial (25x25, todo 0 salvo algunas pistas)
# Puedes cambiar los valores aquí para ver cómo Alice resuelve diferentes configuraciones
tablero = [[0 for _ in range(25)] for _ in range(25)]
tablero[0][0] = 5
tablero[12][12] = 25
tablero[24][24] = 1

# 2. Iniciar Alice como subproceso
# Si tienes problemas con el PATH, sustituye 'julia' por la ruta completa (ej. '/usr/bin/julia')
process = subprocess.Popen(
    ['julia', 'alice.jl'],
    stdin=subprocess.PIPE,
    stdout=subprocess.PIPE,
    text=True
)

# 3. Enviar datos a Alice
print("[BOB] Enviando tablero a Alice...")
process.stdin.write(json.dumps(tablero) + "\n")
process.stdin.flush()

# 4. Leer respuesta de Alice
line = process.stdout.readline()
if line:
    resultado = json.loads(line)
    renderizar_sudoku(resultado)
    print("\n[BOB] ¡Resolución completada!")
else:
    print("[BOB] Error: Alice no respondió.")

# 5. Cerrar proceso
process.kill()

File: bob.js
const { spawn, execSync } = require('child_process');

// Auto-descubrimiento: encuentra la ruta exacta de Julia en tu sistema
const JULIA_PATH = execSync('which julia').toString().trim();

console.log(`[BOB] Julia detectada en: ${JULIA_PATH}`);

// Lanzar Julia
const alice = spawn(JULIA_PATH, ['alice.jl']);

const solicitud = {
    matriz: [[3, 3], [8, 3]],
    vector: [6, 6]
};

// Capturar respuesta
alice.stdout.on('data', (data) => {
    const resultado = JSON.parse(data.toString());
    console.log("[BOB] Resultado recibido:", resultado);
    alice.kill();
    process.exit(0);
});

// Enviar solicitud
alice.stdin.write(JSON.stringify(solicitud) + '\n');

// Manejo de errores
alice.stderr.on('data', (data) => {
    console.error(`[ALICE ERROR]: ${data}`);
});
